pkg/gzip/gzip_test.go

```diff
@@ -47,6 +45,10 @@ func TestFileWithoutDir(t *testing.T) {
 }
 
 func TestSelfHardlink(t *testing.T) {
+	if runtime.GOOS == "darwin" {
+		t.Skip("bsdtar doesn't handle a selflink tarball")
```

Comment:
a little suspicious at this.

The self-hardlink tarballs are only creatable by GNU tars, and only extractable by GNU tars. We have a self-hardlink tarball checked in to `testdata`, and obviously this test will fail if `tar` is a BSD tar on the system running the test.

This test is effectively asking "is a file that is only creatable by GNU tars extractable via the `tar` on the system the test happens to be run on?"

Testing that the tarballs are, in fact, extractable by GNU tar... doesn't really seem to be our job. Maybe we can just delete this test.